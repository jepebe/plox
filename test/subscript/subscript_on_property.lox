class A {
    init(a) {
        this.a = a;
    }
    __get__ (index) {
        return '[' + index * this.a + ']';
    }
}
class B {
    init(b) {
        this.b = A(b);
    }

    property {
        return this.b;
    }
}

var b = B(3.14);
print b.property[1];
print b.property[2];
