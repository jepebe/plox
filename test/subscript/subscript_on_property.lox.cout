[38;5;1m[line 14] Error at '{': Expect '(' after function name.[0m
1 error(s) and 0 warning(s) occurred
