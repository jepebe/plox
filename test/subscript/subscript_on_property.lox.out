[3.14]
[6.28]
